conjure-java-core/src/test/java/com/palantir/conjure/java/UndertowServiceEteTest.java

```diff
@@ -127,6 +128,7 @@ public static void before() {
                 EteBinaryServiceEndpoint.of(new UndertowBinaryResource()));
         endpoints.forEach(endpoint -> endpoint.create(context).register(handler));
         server = Undertow.builder()
+                .setServerOption(UndertowOptions.DECODE_URL, false)
```

Comment:
can we make it so that we get the correct behavior by default?

Reply:
No, undertow decodes URLs internally prior to routing, but excludes slashes (`%2F`) to avoid request smuggling vulnerabilities. We have to ask undertow not to decode until after we've routed requests.

Reply:
Let me try again :) What I would like is the a conjure-undertow-based Witchcraft service does the expected thing, i.e., a request to `GET /foo%2Fbar` for a template `/{var}` gives me `var=="foo/bar"`. Surely we can do this without asking Undertow to change?

Reply:
Of course, we'll make a slight change to how we configure undertow (Add `.setServerOption(UndertowOptions.DECODE_URL, false)` and a URLDecodingHandler), and everything will work fine. Is that what you're asking?

Reply:
I have an idea for an alternative implementation of this change which will likely be less invasive

Reply:
ack

Reply:
My quick attempt at an alternative fell through, it will be a fair amount of work, I think this PR provides what we need.

Reply:
@uschi2000 Any concerns about this PR in its current state?

Reply:
looks ok. the undertow/jersey behavior already treats path parameters correctly, ja?

Reply:
That is correct